drivers/hwinfo/hwinfo_litex.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2019 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/hwinfo.h>
+#include <soc.h>
+#include <string.h>
+#include <device.h>
+#include <sys/util.h>
+
+ssize_t z_impl_hwinfo_get_device_id(u8_t *buffer, size_t length)
+{
+	u32_t volatile *ptr = (u32_t volatile *)(DT_INST_0_LITEX_DNA0_BASE_ADDRESS);
+	ssize_t end = MIN(length, (DT_INST_0_LITEX_DNA0_SIZE / 4));
```

Comment:
Just a little nick-pit:
use `sizeof(u32_t)` instead of the constant 4.

Reply:
Otherwise, it looks good for me. Unfortunately, I don't have any FPGA boards to test.
Did you run the test-cases? There is a shell module to verify the output. Please do so if you haven't already.

Reply:
> Just a little nick-pit:
> use `sizeof(u32_t)` instead of the constant 4.

Changed.

> Otherwise, it looks good for me. Unfortunately, I don't have any FPGA boards to test.
> Did you run the test-cases? There is a shell module to verify the output. Please do so if you haven't already.

I've tested it with `shell_module` demo with `hwinfo_shell` on Arty board on two pieces of HW:
```
uart:~$ hwinfo devid
Length: 8
ID: 0x005c1d2e50210854
```

```
uart:~$ hwinfo devid
Length: 8
ID: 0x005cc54e233ef054
```